pkg/oktaapi/client_wrapper.go

```diff
@@ -0,0 +1,174 @@
+	FirstName   string `json:"firstName"`
+	LastName    string `json:"lastName"`
+	DisplayName string `json:"displayName"`
+	Email       string `json:"email"`
+	Login       string `json:"login"`
+	SourceType  string `json:"SourceType"`
+}
+
+func (cw *ClientWrapper) parseOktaProfileResponse(ctx context.Context, profile *okta.UserProfile) (*oktaUserResponse, error) {
+	logger := appcontext.ZLogger(ctx)
+
+	// Create an okaUserProfile to return
+	parsedProfile := &oktaUserResponse{}
+
+	// Marshal the profile into a string so we can later unmarshal it into a struct
+	responseString, err := json.Marshal(profile)
+	if err != nil {
+		logger.Error("error marshalling okta response", zap.Error(err))
+		return nil, err
+	}
+
+	// Unmarshal the string into the oktaUserProfile type
+	err = json.Unmarshal(responseString, parsedProfile)
+	if err != nil {
+		logger.Error("error unmarshalling okta response", zap.Error(err))
+		return nil, err
+	}
+
+	return parsedProfile, nil
+}
+
+func (o *oktaUserResponse) toUserInfo() *models.UserInfo {
+	return &models.UserInfo{
+		CommonName: o.DisplayName,
+		Email:      models.NewEmailAddress(o.Email),
+		EuaUserID:  o.Login,
+		// DisplayName: o.DisplayName,
+		// Email:       o.Email,
+		// Username:    o.Login,
+		// FirstName:   o.FirstName,
+		// LastName:    o.LastName,
+	}
+}
+
+// FetchUserInfos fetches users from Okta by EUA ID
+func (cw *ClientWrapper) FetchUserInfos(ctx context.Context, usernames []string) ([]*models.UserInfo, error) {
+	logger := appcontext.ZLogger(ctx)
+	users := []*models.UserInfo{}
+	if len(usernames) == 0 {
+		return users, nil
+	}
+
+	var euaSearches []string
+	for _, username := range usernames {
+		euaSearches = append(euaSearches, fmt.Sprintf(`profile.login eq "%v"`, username))
+	}
+	euaSearch := strings.Join(euaSearches, " or ")
+	searchString := fmt.Sprintf(`(%v)`, euaSearch)
+	search := query.NewQueryParams(query.WithSearch(searchString))
+
+	searchedUsers, _, err := cw.oktaClient.User.ListUsers(ctx, search)
+	if err != nil {
+		logger.Error("Error searching Okta users", zap.Error(err), zap.String("usernames", strings.Join(usernames, ", ")))
+		return nil, err
+	}
+
+	for _, user := range searchedUsers {
+		profile, err := cw.parseOktaProfileResponse(ctx, user.Profile)
+		if err != nil {
+			return nil, err
+		}
+		users = append(users, profile.toUserInfo())
+	}
+	return users, nil
+}
+
+// FetchUserInfo fetches a single user from Okta by EUA ID
+func (cw *ClientWrapper) FetchUserInfo(ctx context.Context, username string) (*models.UserInfo, error) {
+	logger := appcontext.ZLogger(ctx)
+
+	user, _, err := cw.oktaClient.User.GetUser(ctx, username)
+	if err != nil {
+		logger.Error("Error fetching Okta user", zap.Error(err), zap.String("username", username))
+		return nil, err
+	}
+
+	profile, err := cw.parseOktaProfileResponse(ctx, user.Profile)
+	if err != nil {
+		return nil, err
+	}
+
+	return profile.toUserInfo(), nil
+}
+
+// SearchCommonNameContains is a wrapper for SearchByName
```

Comment:
Why is this wrapper method needed? It doesn't look like `SearchByName()` is called from anywhere else.

Reply:
This is probably my fault here from pulling this over from MINT. MINT names the function differently, and instead of renaming the method I wrapped it so the core functionality between the apps is unchanged. This is probably unimportant, but that's more or less why it's like that.

Reply:
Addressed in [6420f51](https://github.com/CMSgov/easi-app/pull/2046/commits/6420f51a448742a7f3a31ce0745e6ab4428c5c78)